projects/storefrontlib/cms-components/myaccount/payment-methods/payment-methods.component.ts

```diff
@@ -96,24 +96,28 @@ export class PaymentMethodsComponent implements OnInit {
   }
 
   deletePaymentMethod(paymentMethod: PaymentDetails): void {
-    this.userPaymentService.deletePaymentMethod(paymentMethod.id);
-    this.editCard = null;
+    if (paymentMethod.id) {
+      this.userPaymentService.deletePaymentMethod(paymentMethod.id);
+      this.editCard = undefined;
+    }
   }
 
   setEdit(paymentMethod: PaymentDetails): void {
     this.editCard = paymentMethod.id;
   }
 
   cancelCard(): void {
-    this.editCard = null;
+    this.editCard = undefined;
   }
 
   setDefaultPaymentMethod(paymentMethod: PaymentDetails): void {
-    this.userPaymentService.setPaymentMethodAsDefault(paymentMethod.id);
-    this.globalMessageService?.add(
-      { key: 'paymentMessages.setAsDefaultSuccessfully' },
-      GlobalMessageType.MSG_TYPE_CONFIRMATION
-    );
+    if (paymentMethod.id) {
+      this.userPaymentService.setPaymentMethodAsDefault(paymentMethod.id);
+      this.globalMessageService?.add(
+        { key: 'paymentMessages.setAsDefaultSuccessfully' },
+        GlobalMessageType.MSG_TYPE_CONFIRMATION
+      );
+    }
```

Comment:
I think that's probably ok because the payment method id should probably not ever be falsy if this function is called.
I was wondering however if we should throw an exception or something like this instead of silently doing nothing.  Let's discuss.

Reply:
If there is "PaymentDetails", it must have id. (like Product, if there is product, it must have code). But, we defined `id` as optional. So, customers can call it `setDefaultPaymentMethod({})`.